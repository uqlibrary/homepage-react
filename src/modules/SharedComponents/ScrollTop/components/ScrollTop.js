import React from 'react';
import PropTypes from 'prop-types';
import Fab from '@mui/material/Fab';
import Hidden from '@mui/material/Hidden';
import withStyles from '@mui/styles/withStyles';
import ArrowUpwardIcon from '@mui/icons-material/ArrowUpward';

export const styles = () => ({
    scrollTop: {
        opacity: 0,
        position: 'absolute',
        right: 32,
        bottom: 16,
        '-webkit-transition': 'opacity 0.5s',
        transition: 'opacity 0.3s',
        '&hover:': {
            '-webkit-transition': 'opacity 0.5s',
            transition: 'opacity 0.3s',
            opacity: 0.9,
        },
    },
});

/* istanbul ignore next */
const scrollWindowToTop = event => {
    event.preventDefault();
    document.getElementById('content-container').scrollTop = 0;
};

export const ScrollTop = ({ show, showAfter, classes }) => {
    /* istanbul ignore next */
    const scrollableContainer = document.getElementById('content-container');
    /* istanbul ignore next */
    if (
        !!show &&
        !!scrollableContainer &&
        !!document &&
        !!document.getElementById('scrolltopbtn') &&
        !!document.getElementById('scrolltopbtn').style
    ) {
        scrollableContainer.onscroll = () => {
            if (scrollableContainer.scrollTop > showAfter) {
                document.getElementById('scrolltopbtn').style.opacity = '0.5';
                document.getElementById('scrolltopbtn').style.right = '32px !important';
            } else {
                document.getElementById('scrolltopbtn').style.opacity = '0';
                document.getElementById('scrolltopbtn').style.right = '-1000px !important';
            }
        };
    }
    if (!!show) {
        return (
            <Hidden mdDown>
                <Fab
                    color="secondary"
                    aria-label="Scroll to top of page"
                    className={classes.scrollTop}
                    id="scrolltopbtn"
                    title="Scroll to top of page"
                    onClick={scrollWindowToTop}
                >
                    <ArrowUpwardIcon />
                </Fab>
            </Hidden>
        );
    } else {
        return <div className={'scrolltop-hidden'} />;
    }
};

ScrollTop.propTypes = {
    show: PropTypes.bool,
    showAfter: PropTypes.number,
};

ScrollTop.defaultProps = {
    show: false,
    showAfter: 100,
};

export default withStyles(styles, { withTheme: false })(ScrollTop);
