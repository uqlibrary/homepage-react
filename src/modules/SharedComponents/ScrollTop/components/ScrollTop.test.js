import { ScrollTop as ScrollTopClass } from './ScrollTop';
import ScrollTop from './ScrollTop';

function setup(testProps, isShallow = false, requiresStore = false, context = {}) {
    const props = {
        classes: {},
        ...testProps,
    };
    return getElement(ScrollTopClass, props, isShallow, requiresStore, context);
}

describe('Component ScrollTop', () => {
    it('should render component', () => {
        const wrapper = setup({ show: true });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should not render component', () => {
        const wrapper = setup({ show: false });
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

function setupFull(testProps, isShallow = false, requiresStore = false, context = {}) {
    const props = {
        classes: {},
        ...testProps,
    };
    return getElement(ScrollTop, props, isShallow, requiresStore, context);
}

describe('Component ScrollTop', () => {
    it('should render component', () => {
        const wrapper = setupFull({ show: true });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should not render component', () => {
        const wrapper = setupFull({ show: false });
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
