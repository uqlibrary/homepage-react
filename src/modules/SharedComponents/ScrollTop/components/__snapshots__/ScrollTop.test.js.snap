// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ScrollTop should not render component 1`] = `
<div
  className="scrolltop-hidden"
/>
`;

exports[`Component ScrollTop should not render component 2`] = `
<ScrollTop
  classes={
    Object {
      "scrollTop": "ScrollTop-scrollTop-1",
    }
  }
  show={false}
  showAfter={100}
/>
`;

exports[`Component ScrollTop should render component 1`] = `
<Hidden
  smDown={true}
>
  <WithStyles(ForwardRef(Fab))
    aria-label="Scroll to top of page"
    color="secondary"
    id="scrolltopbtn"
    onClick={[Function]}
    title="Scroll to top of page"
  >
    <Memo(ForwardRef(ArrowUpwardIcon)) />
  </WithStyles(ForwardRef(Fab))>
</Hidden>
`;

exports[`Component ScrollTop should render component 2`] = `
<ScrollTop
  classes={
    Object {
      "scrollTop": "ScrollTop-scrollTop-1",
    }
  }
  show={true}
  showAfter={100}
/>
`;
